src/Voron/Impl/SliceSmallSet.cs

```diff
@@ -226,10 +218,10 @@ public void Clear()
 
         public void Dispose()
         {
-            _perCorePools.KeySizesPool.Return(_keySizes, clearArray: true);
-            _perCorePools.KeyHashesPool.Return(_keyHashes, clearArray: true);
-            _perCorePools.KeysPool.Return(_keys, clearArray: true);
-            _perCorePools.ValuesPool.Return(_values, clearArray: true);
+            _perCorePools.KeySizesPool.Return(_keySizes);
+            _perCorePools.KeyHashesPool.Return(_keyHashes);
+            _perCorePools.KeysPool.Return(_keys);
+            _perCorePools.ValuesPool.Return(_values);
```

Comment:
Problem is that `_values` is used to `IPrepareForCommit` instances.
That means that if we _don't_ clear them, the instance will remain in memory. 
For example, a `Lookup<T>` which holds a reference to `LowLevelTransaction`, etc.

A good solution for that without adding additional cost would be to change `ReturnValues` to `ConsumeValues`, so it will fetch the value _and reset the instance to `null`.

This way, we don't pay the cost of zeroing the memory, but won't retain the instances.

Reply:
Makes sense. The other alternative is to just do that on `_values` but explicitly in order to use the `_currentIdx` value instead and do not do wasted work.